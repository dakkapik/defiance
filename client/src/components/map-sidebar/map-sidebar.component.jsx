import React, { useState } from "react";

import { connect } from "react-redux";

import "./map-sidebar.styles.scss";
//Actions
import {
  ordersSocketOn,
  ordersSocketOff,
} from "../../redux/orders/orders.action";
import { clearActiveDriver } from "../../redux/drivers/drivers.action";
import { socketOff } from "../../redux/socket/socket.action";
//Components
import ArrowModalButton from "../arrow-modal-button/arrow-expanded-modal-button.component";
import Map from "../map/map.component";
import SaveModalButton from "../save-modal-button/save-modal-button.component";
import Button from "@material-ui/core/Button";
import StoreList from "../store-list/store-list.component";
import Orders from "../orders/orders.component";
//Assets
import arrow from "./arrow.png";
import DynamicDriverList from "../dynamic-driverlist/dynamic-driverlist.component";
import { DisconnectButtonStyles } from "./map-sidebar.styles";
/*
MapSideBar functionality
Renders: Map componenet always 
Renders components conditionally based on when the store buttons are clicked and the arrow button:
StoreList: conditionally
Dynamic Driver: conditionally
Orders: conditionally
*/

export const MapSideBar = ({
  //comparison variables for no question on red arrow closing if no changes
  apiorders,
  unassigned_orders,
  socket,
  showorders,
  socketOff,
  clearActiveDriver,
  ordersSocketOn,
  ordersSocketOff,
}) => {
  const [show_arrow_modal, openArrowModal] = useState(false);
  const disconnect_button_classes = DisconnectButtonStyles();
  const handleClose = () => {
    openArrowModal(false);
  };

  const handleOpenArrowModal = () => {
    openArrowModal(true);
  };

  return (
    <div className="map-side-container">
      <Map />
      {/* 
       If Manager does click a store then socket=true  then dynamic driver loads with showorders being false
       */}
      {socket ? (
        <div className="sidebar-container">
          <div className={showorders ? "side-bar-expanded " : "side-bar"}>
            {showorders ? (
              <div className="top-container-expanded">
                <div className="top-container-expanded__left-section" />
                <div className="top-container-expanded__right-section">
                  <SaveModalButton />
                </div>
              </div>
            ) : (
              <div className="top-container">
                <Button
                  classes={{
                    label: disconnect_button_classes.label,
                    root: disconnect_button_classes.root,
                  }}
                  variant="outlined"
                  color="secondary"
                  onClick={() => {
                    socketOff(false);
                    clearActiveDriver();
                  }}
                >
                  Disconnect
                </Button>
              </div>
            )}

            {/*If Manager clicks the arrow then showorders is true then Order Component renders */}
            {showorders ? <Orders /> : <DynamicDriverList />}
          </div>
          <ArrowModalButton
            show_arrow_modal={show_arrow_modal}
            handleClose={handleClose}
          />
          {showorders ? (
            <img
              src={arrow}
              className="arrow-expanded"
              alt="expanded-arrow"
              onClick={() => {
                /*
               if no orders where dragged to a driver then don't popup a modal and go to full map screen   
              */

                apiorders.length === unassigned_orders.length
                  ? ordersSocketOff()
                  : handleOpenArrowModal();
              }}
            />
          ) : (
            <img
              src={arrow}
              alt="arrow"
              onClick={() => {
                ordersSocketOn();
              }}
              className="arrow"
            />
          )}
        </div>
      ) : (
        //If Manager does not click store socket=false then  StoreList component renders
        <div className="side-bar">
          <StoreList />
        </div>
      )}
    </div>
  );
};

const mapStateToProps = (state) => ({
  apiorders: state.orders.apiorders,
  unassigned_orders: state.orders.currentdragdrop.columns["column-1"].orderIds,
  socket: state.socket.socketToggle,
  showorders: state.orders.showorders,
});

const mapDispatchToProps = (dispatch) => ({
  ordersSocketOn: () => dispatch(ordersSocketOn()),
  ordersSocketOff: (x) => dispatch(ordersSocketOff(x)),
  socketOff: (bool) => dispatch(socketOff(bool)),
  clearActiveDriver: () => dispatch(clearActiveDriver()),
});

export default connect(mapStateToProps, mapDispatchToProps)(MapSideBar);
