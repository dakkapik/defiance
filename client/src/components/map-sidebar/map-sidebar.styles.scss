@import "../../global-css/variables.scss";
.map-side-container {
  position: relative;
}

.top-container {
  display: flex;
  padding-top: 10px;
  padding-bottom: 20px;
  justify-content: space-around;
}
.top-container-expanded {
  display: flex;
  padding-top: 10px;
  padding-bottom: 20px;
  &__left-section {
    display: flex;
    justify-content: center;
    // background-color: blue;
    margin: 1vh;
    width: 50%;
  }
  &__right-section {
    // background-color: red;
    margin: 1vh;
    display: flex;
    width: 50%;
    justify-content: center;
    align-items: center;
  }
}
.side-bar {
  width: 15em;
  height: 95vh;
  color: white;
  background-color: $lightdark;
  position: absolute;
  top: 0;
  right: 0;
  overflow: auto;
  transition: width 1s;

  @media screen and (max-width: 670px) {
    position: relative;
    transition: unset;
    width: unset;
    height: fit-content;
  }
}

.arrow {
  position: absolute;
  height: 30px;
  width: 30px;
  top: 45%;
  right: 14em;
  transition: right 1s;
  -webkit-transform: scaleX(-1);
  transform: scaleX(-1);
  cursor: pointer;
  @media screen and (max-width: 670px) {
    display: none;
  }
}

// @keyframes rotation {
//   from {
//     transform: rotate(180deg);
//   }
//   to {
//     transform: rotate(360deg);
//   }
// }

.arrow-expanded {
  // animation: rotation 0.5s linear;

  cursor: pointer;
  position: absolute;
  // -webkit-transform: scaleX(-1);
  // transform: scaleX(-1);
  height: 30px;
  width: 30px;
  top: 45%;
  right: 59%;
  transition: right 1s;

  @media screen and (max-width: 1150px) {
    right: 25em;
  }

  @media screen and (max-width: 670px) {
    display: none;
  }
}
.side-bar-expanded {
  width: 60.5%;
  transition: width 1s;
  height: 95vh;
  color: white;
  background-color: $lightdark;
  position: absolute;
  top: 0;
  right: 0;
  overflow: auto;

  @media screen and (max-width: 1150px) {
    width: 26em;
  }

  @media screen and (max-width: 670px) {
    position: relative;

    width: unset;
    height: fit-content;
  }
  &__driverorder-container {
    padding: 10px;
    display: flex;
    justify-content: space-evenly;

    // border-bottom: 1px dotted white;
  }
}
