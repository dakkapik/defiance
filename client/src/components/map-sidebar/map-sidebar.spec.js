import React from "react";
import { MapSideBar } from "./map-sidebar.component";
import { shallow } from "enzyme";

describe("Orders component render", () => {
  let wrapper;
  let mockSocketOff;
  let mockclearActiveDriver;
  let mockordersSocketOn;
  let mockordersSocketOff;

  beforeEach(() => {
    mockSocketOff = jest.fn();
    mockclearActiveDriver = jest.fn();
    mockordersSocketOn = jest.fn();
    mockordersSocketOff = jest.fn();
    const mockProps = {
      socketOff: mockSocketOff,
      clearActiveDriver: mockclearActiveDriver,
      ordersSocketOn: mockordersSocketOn,
      ordersSocketOff: mockordersSocketOff,
      socket: false,
      showorders: false,
    };

    wrapper = shallow(<MapSideBar {...mockProps} />);
  });

  it("should render MapSideBar component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
