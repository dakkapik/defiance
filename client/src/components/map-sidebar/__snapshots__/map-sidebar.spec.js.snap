// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders component render should render MapSideBar component 1`] = `ShallowWrapper {}`;
